### PR TITLE
feat: add asset types

### PR DESCRIPTION
**Pull request by fruizrob**
This PR adds support for the following asset types and associated documentation:

| Asset | Type definition |
|:------|:-----------|
| Fully Qualified Domain Name | `FQDN` |
| Autonomous System | `AutonomousSystem` |
| Regional Internet Registry Organization | `RIROrganization` |
| IP Address | `IPAddress` |
| Netblock | `Netblock` |

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,28 @@
...
+## Assets
+
+Supported asset types:
+
+| Asset | Type definition |
+|:------|:-----------|
+| Fully Qualified Domain Name | `FQDN` |
+| Autonomous System | `AutonomousSystem` |
+| Regional Internet Registry Organization | `RIROrganization` |
+| IP Address | `IPAddress` |
```
Comment (lonefoxzero) [inline README.md:16]:
```suggestion
| Internet Protocol Address | `IPAddress` |
```

File: domain/domain.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Includes several types
+// that represent common domain-related
+// concepts, such as FQDN for representing
+// fully qualified domain names.
+//
+// These types can be used to store
+// and manipulate domain-related data
+// in Go programs.
+package domain
```
Comment (lonefoxzero) [inline domain/domain.go:9]: I'm curious - why does this empty file exist?

**Review by fnmendez — COMMENTED** [review]
shouldn’t those `example_test.go` files be renamed to `domain_test.go` and `network_test.go`?

File: README.md [hunk 3]
```diff
@@ -3,3 +3,28 @@
...
+| Netblock | `Netblock` |
+
+## Documentation
+
+The documentation can be found in the Go packages repository:[open-asset-model](https://pkg.go.dev/github.com/owasp-amass/open-asset-model), there you can find the reference and descriptions to the types.
+
+## Contributing [![Contribute Yes](https://img.shields.io/badge/contribute-yes-brightgreen.svg)](./CONTRIBUTING.md) [![Chat on Discord](https://img.shields.io/discord/433729817918308352.svg?logo=discord)](https://discord.gg/HNePVyX3cp)
```
Comment (rynmrtn) [inline README.md]: move these up to the top w/ the other GoDoc badge

File: network/network.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// Provides several types
```
Comment (rynmrtn) [inline network/network.go]: rename to doc.go

File: network/netblock_test.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			prefix, _ := netip.ParsePrefix(tt.cidr)
+			netblock := Netblock{Cidr: prefix, Type: tt.netType}
+
+			jsonData, err := json.Marshal(netblock)
+
+			assert.NoError(t, err)
+			assert.Contains(t, string(jsonData), fmt.Sprintf(`"cidr":"%s"`, tt.cidr))
+			assert.Contains(t, string(jsonData), fmt.Sprintf(`"type":"%s"`, tt.netType))
```
Comment (rynmrtn) [inline network/netblock_test.go]: please use testify's [require.JSONEq](https://pkg.go.dev/github.com/stretchr/testify@v1.8.2/require#JSONEq) for a stronger assertion

**Review by rynmrtn — COMMENTED** [review]
Thank you for adding these asset types @fruizrob! 

There are a few things to address:

- [ ] Rename network/network.go to network/doc.go, as it's primarily serving as documentation.
- [ ] Switch all JSON assertions to rely upon require.JSONEq
- [ ] Update the readme badges to the top w/ the GoDoc one

File: README.md (hunk 3 above)
Comment (fruizrob) [reply README.md]: great idea, done 👍

File: network/network.go (hunk 4 above)
Comment (fruizrob) [reply network/network.go]: sg, I also changed `domain/domain.go` to `domain/doc.go`

File: network/netblock_test.go (hunk 5 above)
Comment (fruizrob) [reply network/netblock_test.go]: done, and I changed all the assertions to `testify/require`

**Review by rynmrtn — APPROVED** [review]
thanks for the cleanup commits! Looks really good :)

**Pull request merged**